M3 and M2 Armor Rebalance

Diegoflores31 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->

## About The Pull Request

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->
This PR aims to make Armor be a little different and to be able to notice the difference betwen types of armor 
(light,medium,heavy) and (M2,M3)

_**Light:**_ 
a Chesplate that was stripped of his arm and leg protection to provide the wearer with more speed.

_**Medium**_
Covers your arms and legs keeping you protected against attacks.

_**Heavy**_
Will cover Arms , legs and feet similar to how GL armor does but it will heavily slow you down. Adds 1 more pouch to be as good as medium.

**M2 armor and M3 armor**
Current state of M2 and M3 armor was a little confusing since while M2 is supposed to be an older design it seemed to be better than M3 while contradicting the original description that describes it as a , Chestplate to protect the Chest agains blows and bullets . 

For this reason. M2 armor was adapted to be equivalent of a marine light armor with increased melee and Bullet resistance.
all MPs armor were changed to reflect that they use M2 armor.

Warden and CMP will keep the full body armor while enjoying the increased bullet resistance made by this change.

SO and CO armor was changed to no longer be a child of the MP armor (M2) and instead be a better version of the M3 armor . keeping the light movement speed.

**VC armor**
VC armor now has 3 pouches (same as M3 medium armor) while providing a little more Bomb Protection as this armor is a job locked one should be a little different than standard.

**M4 armor**
RTO and IO had basically the same armor while having 2 different names . this PR eliminates XM4 armor and merges it with M4 armor and gives increased armor to M4 to be a leser B12.
also eliminates Radiation resistance ( no idea why it was even there)

**VL armor**
added 5 bio and 5 rad armor since it provided no resistance at all despite beign arguably heavily more armored than a jacket,

## Why It's Good For The Game

<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding, and may discourage maintainers from reviewing or merging your PR. This section is not strictly required for (non-controversial) fix PRs or backend PRs. -->

This PR aims for giving marines options besides light armor by making it less optimal and providing heavier armor a very much needed improvement and hopefully these changes will provide marines with more diverse gameplay.

Makes M2 armor more clearly differentiated from M3  and tweaks other kinds of armor that were considered "bad" to be a solid option .

## Changelog

<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
balance: Heavy Armor now provides Feet and hand Protection and has the same slots than medium armor.
balance: Heavy Armor values changed to be all on MediumHIGH (25)
balance: Light armor all values reduced to be 5 less than medium armor. (15)
balance: VL vest now provides with 5 bio and 5 radiation armor to be in contrast with jackets.
balance: Standard M2 armor now provides 5 less Melee resistance in exchange of 10 bullet resistance
balance: M2 armor now only has 1 pouch as the description indicates.
balance: CMP and Warden M2 armor have increased values of Bullet armor (HIGH and MediumHIGH respectively)
balance: VC armor now provides one more storage slot and some Explosion resistance.
balance: SO armor is no longer a Child of M2 armor and now inherits M3 Armor but with less speed penalty.
balance: All MP armor is M2 now , no more M2 called M3 and vice versa.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->

carlarctg commented on code/modules/clothing/suits/marine_armor.dm:
```diff
@@ -279,30 +279,25 @@ var/list/squad_colors_chat = list(rgb(230,125,125), rgb(255,230,80), rgb(255,150
 	armor_bomb = CLOTHING_ARMOR_MEDIUM
 	armor_bio = CLOTHING_ARMOR_MEDIUMLOW
 	armor_internaldamage = CLOTHING_ARMOR_MEDIUMLOW
+	flags_armor_protection = BODY_FLAG_CHEST|BODY_FLAG_GROIN
```
the sprite has arms, shoulder, and legs protection, it's going to be very out of place with this

Diegoflores31 replied:
the description quite literally says otherwise

carlarctg commented on code/modules/clothing/suits/marine_armor.dm:
```diff
@@ -342,72 +337,74 @@ var/list/squad_colors_chat = list(rgb(230,125,125), rgb(255,230,80), rgb(255,150
+	desc = "A well-crafted suit of M2 Pattern Armor typically distributed to Wardens. Useful for letting your men know who is in charge."
 	armor_bio = CLOTHING_ARMOR_MEDIUMLOW
 	armor_rad = CLOTHING_ARMOR_MEDIUMLOW
+	flags_armor_protection = BODY_FLAG_CHEST|BODY_FLAG_GROIN|BODY_FLAG_ARMS|BODY_FLAG_LEGS
```
why is it better than normal mp armor??

Diegoflores31 replied:
because it is "well crafted " 3:)

carlarctg commented on code/modules/clothing/suits/marine_armor.dm:
```diff
@@ -342,72 +337,74 @@ var/list/squad_colors_chat = list(rgb(230,125,125), rgb(255,230,80), rgb(255,150
+	desc = "A well-crafted suit of M2 Pattern Armor typically distributed to Chief MPs. Useful for letting your men know who is in charge."
 	uniform_restricted = list(/obj/item/clothing/under/marine/officer/warrant)
-	specialty = "M3 pattern chief MP"
+	flags_armor_protection = BODY_FLAG_CHEST|BODY_FLAG_GROIN|BODY_FLAG_ARMS|BODY_FLAG_LEGS
```
same thing here, the stats aren't different normally so why is it stronger

Diegoflores31 replied:
its a special issued M2 as it says on the description

Diegoflores31 replied:
also its a different sprite

carlarctg commented on code/modules/clothing/suits/marine_armor.dm:
```diff
@@ -510,8 +507,8 @@ var/list/squad_colors_chat = list(rgb(230,125,125), rgb(255,230,80), rgb(255,150
-	armor_bio = CLOTHING_ARMOR_NONE
-	armor_rad = CLOTHING_ARMOR_NONE
+	armor_bio = CLOTHING_ARMOR_LOW
+	armor_rad = CLOTHING_ARMOR_LOW
```
undocumented

Diegoflores31 replied:
gonna add it now . i forgot

carlarctg commented on code/modules/clothing/suits/marine_armor.dm:
```diff
@@ -492,12 +488,13 @@ var/list/squad_colors_chat = list(rgb(230,125,125), rgb(255,230,80), rgb(255,150
+	armor_bio = CLOTHING_ARMOR_MEDIUM
+	armor_rad = CLOTHING_ARMOR_MEDIUMLOW
 	armor_internaldamage = CLOTHING_ARMOR_LOW
+	flags_armor_protection = BODY_FLAG_CHEST|BODY_FLAG_GROIN
```
same as MP armor, doesnt change sprites so it's going to look out of place.  description should probs be edited too

Diegoflores31 replied:
light armor sprite is just a chestplate

carlarctg commented on code/modules/clothing/suits/marine_armor.dm:
```diff
@@ -560,7 +557,7 @@ var/list/squad_colors_chat = list(rgb(230,125,125), rgb(255,230,80), rgb(255,150
 	armor_rad = CLOTHING_ARMOR_MEDIUMHIGH
 	armor_internaldamage = CLOTHING_ARMOR_MEDIUMHIGH
-	storage_slots = 2
+	flags_armor_protection = BODY_FLAG_CHEST|BODY_FLAG_GROIN|BODY_FLAG_ARMS|BODY_FLAG_LEGS|BODY_FLAG_FEET
```
undocumentedly sets storage slots to 3

Diegoflores31 replied:
forgot about it . gonna add it asap

carlarctg commented:
bah why doesn't it show up to the definition so you can see what object im talking about >:(

Diegoflores31 commented on code/modules/clothing/suits/marine_armor.dm:
```diff
@@ -279,30 +279,25 @@ var/list/squad_colors_chat = list(rgb(230,125,125), rgb(255,230,80), rgb(255,150
 	icon_state = "io"
 	name = "\improper M4 pattern marine armor"
-	desc = "A well tinkered and crafted hybrid of Smart-Gunner mesh and M3 pattern plates. Robust, yet nimble, with room for all your pouches. Required for carrying a Radio Telephone Pack."
+	desc = "An update to the aging M3 vest, M4 armor provides better protection against high-velocity impacts than its predecessor without sacrificing mobility."
```
the name is actually M4X in the lore but M4 sounds just better and simpler

Diegoflores31 replied:
last description did not talked about the brigthness just says it has "pouches" . will see what i can do to add something talking about the lights

stanalbatross reviewed (changes requested):
architect changes :
`balance: Heavy Armor now provides Feet Protection and has the same slots than medium armor.
balance: Light armor only covers Chest and groin now.`
remove. This is not the right way to make these armours "different" from each other as it makes light armour COMPLETELY useless (almost zero armour on arms and legs because gloves/boots only cover hands/feet). Heavy armour buff is misguided (xenos will just aim hand). Little elaboration on what benefit this actually provides to the main gameplay loop other than "makes other armour better!" (aka forces people to pick it because of light being made incredibly shit). Also light has protection on arms/legs on the sprite

`balance: Standard M2 armor now provides only protection to the Chest and Groin as Advertised (excepting Warden and CMP).`
Also remove, makes these armours totally useless in combat + as carlarc said, it has protection on the sprite. Update the desc instead, you have completely misinterpreted the armour's purpose.

`balance: Deleted IO armor and Merged it with RTO armor , making it just M4 armor with a new description.`
Makes zero IC sense (why would RTOs and IOs be issued the same armour?), IO armour should instead be rebalanced to be somewhat faster RTO armour. Without this people will just keep ditching it.

`refactor: All MP armor is M2 now , no more M2 called M3 and vice versa.`
this is not a refactor

stanalbatross reviewed (changes requested):
also update your pr title it's non descriptive

Diegoflores31 commented:
> `balance: Deleted IO armor and Merged it with RTO armor , making it just M4 armor with a new description.` Makes zero IC sense (why would RTOs and IOs be issued the same armour?), IO armour should instead be rebalanced to be somewhat faster RTO armour. Without this people will just keep ditching it.

RTO armor and IO armor is codewise the same armor . has no different sprite and not even a different description . its just complete bloat . IO armor doesnt even exist in the lore since the one that exists is called M4X that i reduced to M4 for an easier name in the fusion made on this PR . 

Additionally this armor was buffed in this PR to be somewhat better than M3 making it more viable than standard M3.

Diegoflores31 commented:
**UPDATE**
M2 armor covers Arms and legs again.
Base MP M2 armor got melee armor reduced by 5 but bullet increased by 5
M2 armor description Changed to display it actually covers limbs 
Warden and CMP armor description edited to show changes to their Bullet resistance.
Warden and CMP armor got increased Bullet resistance compared from normal M2 armor
M3 Heavy armor , got all starts reduced to "MediumHIGH" but instead received full body protection.
M3 Light armor covers Arms and Legs again but got all his stats reduced to be 5 less than medium armor.

stanalbatross reviewed (changes requested):
balance: Deleted IO armor and Merged it with RTO armor , making it just M4 armor with a new description.
Makes zero IC sense (why would RTOs and IOs be issued the same armour?), IO armour should instead be rebalanced to be somewhat faster RTO armour. Without this people will just keep ditching it.

You haven't done this change

Diegoflores31 commented:
**UPDATE**
Dropped all changes on M4 armor and reverted it as it was before . Making more fast armors just goes against the intention of this Pr in general so it will be just left as is.

Wintermote commented:
IO's don't need more protection, they need speed. You probably do more running around the map as an IO than any other marine. In my experience, IO players just immediately ditch their armor for light armor for this and a few other reasons.

IMO just make it the same stats as light armor but with 3 or 4 slots instead of 2
